libbeat/docs/security/users.asciidoc

```diff
@@ -0,0 +1,295 @@
+
+//TODO: Redirect beats-user-access to feature-roles to prevent 404s.
+
+// REVIEWERS: Our built-in roles don't provide enough granularity for some users,
+// and in some cases, the roles don't cover all the tasks that a single user
+// might perform (such as running the setup). Plus the roles don't grant access
+// to Beats indices. I list privileges here (rather than roles) so that the person
+// setting up security has more flexibility and the knowledge they need to
+// create their own roles, if necessary. Is this a likely scenario?
+
+You can use role-based access control to grant {beatname_uc} users access to
+secured resources. The roles that you set up depend on your organization's
+security requirements and the minimum privileges required to use specific
+features. 
+
+{beatname_uc} users typically perform these main roles: they do the initial
+setup, publish monitoring information, and publish events. If they're using
+{kib}, they view and sometimes create visualizations that access {beatname_uc}
+indices.
+
+To assign the most restrictive privileges to users, create a role for each
+purpose. The easiest way to create roles is by using the
+*Management > Security > Roles* UI in {kib}.
+
+Also consider using the <<built-in-beats-roles,built-in roles>> provided by
+{security}.
+
+Want to learn more about creating users and roles? See
+{stack-ov}/elasticsearch-security.html[Securing the {stack}]. Also see
+{stack-ov}/security-privileges.html[Security privileges] for a description of
+available privileges.
+
+// REVIEWERS: I had to grant `monitor` to users to avoid errors, even though I
+// didn't enable monitoring. Is this expected? I've added `monitor` to the
+// privileges described below. Please let me know how I should handle this in
+// the docs.
+
+[[privileges-to-setup-beats]]
+==== Privileges needed for initial setup
+
+Users who set up {beatname_uc} typically need to load mappings, dashboards, and
+other objects used to index data into {es} and visualize it in {kib}. The
+privileges required depend on the setup tasks users need to perform:
+
+[options="header"]
+|====
+|Task | Required privileges
+
+|Set up index templates
+|`manage_index_templates` and `monitor` on cluster
+
+ifndef::no_dashboards[]
+.2+|Set up example dashboards
+|`monitor` on cluster
+
+|`index` on `.kibana*` indices
+endif::no_dashboards[]
+
+ifdef::has_ml_jobs[]
+.3+|Set up machine learning job configurations 
+|`manage_ml` and `monitor` on cluster
+
+|`read` on +{beat_default_index_prefix}-*+ indices
+
+|`read` and `index` on `.kibana*` indices
+endif::has_ml_jobs[]
+
+ifeval::["{beatname_lc}"=="filebeat"]
+|Set up ingest pipelines
+|`manage_ingest_pipelines` and `monitor` on cluster
+endif::[]
+
+ifndef::no_ilm[]
+|Set up index lifecycle policies
+|`manage_ilm` and `monitor` on cluster
+endif::no_ilm[]
+
+ifdef::has_central_config[]
+|Enroll and manage configurations in Beats central management
+|`all` on `.management-beats` index
```

Comment:
You need the `beats_admin` and the `kibana_user` on the Kibana user and the rest will be OK cc @mattapperson

Reply:
Correct. or superuser/whatever cloud called their admin user role

Reply:
@ph Right, I get that the roles work, but can users create their own role that grants the same privileges as the beats_admin role, or must they use our pre-built role?

Reply:
(sorry just saw your comment below....)

Reply:
No, the user must have the correct role, not just index permissions.

Reply:
@mattapperson Is this a common pattern, or just something we are doing for central management? I might be wrong (I've reached out to the security team for guidance here because I'm lacking real visibility into what customers want here), but it seems like we ought to allow users to create their own role for this.

Reply:
@dedemorton you are not wrong. Just an instance of what should be is not what is. The reason for why things are how they are is complex, but valid, and we dont currently have a workaround

Reply:
I've reached out to the security team for input on this. We might be able to create additional built-in roles that make this easier for folks implementing security.